lib/modules/datasource/rubygems/get.ts

```diff
@@ -182,4 +208,116 @@ export class InternalRubyGemsDatasource extends Datasource {
+    namespace: 'datasource-nexus',
+    key: (nexusEndPoint: string) => `isNexusDataSource:${nexusEndPoint}`,
+    ttlMinutes: 1440,
+  })
+  async isNexusDataSource(nexusEndPoint: string): Promise<boolean> {
+    const statusEndPoint = '/service/rest/v1/status';
+    const nexusUrl = nexusEndPoint.concat(statusEndPoint);
+    let isNexus = false;
+    try {
+      const response = await this.http.getJson<HttpResponse>(nexusUrl);
+      isNexus = response.headers?.server?.includes('Nexus') ?? false;
+    } catch (err) {
+      return false;
+    }
+    return isNexus;
+  }
+
+  private async getReleasesFromNexus(
+    repository: string,
+    endPoint: string,
+    dependencyName: string
+  ): Promise<Release[]> {
+    const result: Release[] = [];
+    const gemItems: NexusGemsItems[] = [];
+    const url = this.getNexusGemReleasesEndPoint(
+      repository,
+      endPoint,
+      dependencyName
+    );
+    try {
+      const nexusResponse = (await this.http.getJson<NexusGems>(url))?.body || {
+        continuationToken: null,
+        items: [],
+      };
+      gemItems.push(...nexusResponse.items);
+      const continuationToken: string | null = nexusResponse.continuationToken;
+      if (continuationToken) {
+        const paginationGems = await this.paginationResult(
+          url,
+          continuationToken
+        );
+        gemItems.push(...paginationGems);
+      }
+    } catch (error) {
+      logger.debug('Failed to retreive gem from nexus');
+      return [];
+    }
+
+    result.push(
+      ...gemItems.map((item) => {
+        return { version: item.version, rubyPlatform: 'nexus' };
+      })
+    );
+    return result;
+  }
+
+  private getNexusGemReleasesEndPoint(
+    repository: string,
+    endPoint: string,
+    depName: string
+  ): string {
+    const nexusSearchPoint = '/service/rest/v1/search';
+    const query = `${nexusSearchPoint}?repository=${repository}&name=${depName}&sort=version`;
+    const gemReleasesEndPoint = endPoint.concat(query);
+    return gemReleasesEndPoint;
+  }
+
+  private async paginationResult(
+    url: string,
+    continuationToken: string
+  ): Promise<NexusGemsItems[]> {
+    const gemItems: NexusGemsItems[] = [];
+    let pageToken: string | null = continuationToken;
+    while (pageToken) {
+      const gemsOfPage: NexusGems = await this.pageOfNexusGems(url, pageToken);
+      gemItems.push(...gemsOfPage.items.filter(Boolean));
+      pageToken = gemsOfPage.continuationToken;
+    }
+    return gemItems;
+  }
+
+  private async pageOfNexusGems(
+    url: string,
+    pageToken: string
+  ): Promise<NexusGems> {
+    let nexusPage: NexusGems = { continuationToken: null, items: [] };
+    try {
+      const paginateUrl = url.concat(`&continuationToken=${pageToken}`);
+      const response = await this.http.getJson<NexusGems>(paginateUrl);
+      nexusPage = response.body;
+    } catch (error) {
+      logger.debug('Failed to retreive gems page from nexus');
+      throw error;
```

Comment:
At which level will this error be caught? And at which log level will it be logged?

Reply:
Will be caught here and logged in debug : 
https://github.com/renovatebot/renovate/blob/97547c1b6de67cd681a0527459fcb921bfb4b6ea/lib/modules/datasource/rubygems/get.ts#L270-L274

Reply:
That's an almost identical error as logged above. And do we never log anything about the error itself? e.g. we wont' know from logs if it's 401, 404, 429, 500, etc?